pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -296,97 +295,89 @@ public void skipEntriesFailed(ManagedLedgerException exception, Object ctx) {
+                final Position finalPosition;
+                if (position == null) {
+                    // this should not happen ideally unless a reset is requested for a time
+                    // that spans beyond the retention limits (time/size)
+                    finalPosition = cursor.getFirstPosition();
+                    if (finalPosition == null) {
+                        log.warn("[{}][{}] Unable to find position for timestamp {}. Unable to reset cursor to first position",
+                                topicName, subName, timestamp);
+                        future.completeExceptionally(
+                                new SubscriptionInvalidCursorPosition("Unable to find position for specified timestamp"));
+                        return;
+                    }
+                    log.info(
+                            "[{}][{}] Unable to find position for timestamp {}. Resetting cursor to first position {} in ledger",
+                            topicName, subName, timestamp, finalPosition);
                 } else {
-                    future.completeExceptionally(new BrokerServiceException(exception));
+                    finalPosition = position;
                 }
-                counter.countDown();
-            }
-        });
 
-        try {
-            counter.await();
-        } catch (Exception e) {
-            log.warn("[{}][{}] message find interrupted ", topicName, subName, e);
-            if (future.isCompletedExceptionally()) {
-                return future;
-            }
-            future.completeExceptionally(e);
-            return future;
-        }
+                if (!isFenced.compareAndSet(false, true)) {
+                    future.completeExceptionally(new SubscriptionBusyException("Failed to fence subscription"));
+                    return;
+                }
 
-        if (future.isCompletedExceptionally()) {
-            return future;
-        }
-        if (result.position == null) {
-            // this should not happen ideally unless a reset is requested for a time
-            // that spans beyond the retention limits (time/size)
-            result.position = cursor.getFirstPosition();
-            if (result.position == null) {
-                log.warn("[{}][{}] Unable to find position for timestamp {}. Unable to reset cursor to first position",
-                        topicName, subName, timestamp);
-                future.completeExceptionally(
-                        new SubscriptionInvalidCursorPosition("Unable to find position for specified timestamp"));
-                return future;
-            }
-            log.info(
-                    "[{}][{}] Unable to find position for timestamp {}. Resetting cursor to first position {} in ledger",
-                    topicName, subName, timestamp, result.position);
-        }
+                dispatcher.disconnect().whenComplete((aVoid, throwable) -> {
```

Comment:
Instead can we directly call [disconnect()](https://github.com/yahoo/pulsar/blob/master/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentSubscription.java#L472) which essentially does the same thing.